src/main/java/io/managed/services/test/client/registry/RegistryClient2.java

```diff
@@ -0,0 +1,45 @@
+package io.managed.services.test.client.registry;
+
+import com.openshift.cloud.api.registry.instance.models.RoleMapping;
+import com.openshift.cloud.api.registry.instance.models.ArtifactMetaData;
+import com.openshift.cloud.api.registry.instance.models.ContentCreateRequest;
+
+import com.openshift.cloud.api.registry.instance.groups.item.artifacts.ArtifactsRequestBuilder;
+import io.apicurio.rest.client.auth.exception.NotAuthorizedException;
+import io.managed.services.test.client.BaseApi;
+import io.managed.services.test.client.exception.ApiGenericException;
+import io.managed.services.test.client.exception.ApiUnknownException;
+
+import com.openshift.cloud.api.registry.instance.ApiClient;
+
+import java.net.HttpURLConnection;
+import java.util.HashMap;
+import java.util.concurrent.TimeUnit;
+
+public class RegistryClient2 extends BaseApi {
+
+    private final ApiClient apiClient;
+
+    public RegistryClient2(ApiClient apiClient, String offlineToken) {
+        super(offlineToken);
+        this.apiClient = apiClient;
+    }
+
+    @Override
+    protected ApiUnknownException toApiException(Exception e) {
+        if (e instanceof com.microsoft.kiota.ApiException) {
+            var err = (com.microsoft.kiota.ApiException) e.getCause();
+            return new ApiUnknownException(err.getMessage(), err.hashCode(), new HashMap<>(), "", err);
+        }
+
+        return null;
+    }
+
+    public ArtifactMetaData createArtifact(String groupId, String artifactId, ContentCreateRequest data) throws ApiGenericException {
+          return retry(() -> apiClient.groups(groupId).artifacts().post(data).get(10, TimeUnit.SECONDS));
```

Comment:
RegistryClient implementation using kiota sdk, getting `404` when using this to create `Artifact`.
Not able to figure out if the correct methods that should be called here.

Reply:
I'm not sure exactly where this fails but I can create artifacts with this code:

```java
        var content = new com.openshift.cloud.api.registry.instance.models.ContentCreateRequest();
        content.setContent("{}");
        if (groupId == null) {
            groupId = "default";
        }
        client
                .groups(groupId)
                .artifacts()
                .post(content, config -> config.headers.add("X-Registry-ArtifactType", "JSON"))
                .get(5, TimeUnit.SECONDS);
```

let me know if you can get around the issue!

Reply:
My changes were missing the headers part, but the issue still remains.

Reply:
if you are getting 404 probably you are using an incorrect URL, can you verify by comparing the result of doing the same operation with `curl` for example?

Reply:
confirmed as resolved by Andrea ✅